services/notification/src/helpers/database-helper.js

```diff
@@ -15,19 +16,158 @@ class DatabaseHelper {
     if (cachedDb && cachedDb.serverConfig.isConnected()) {
       this.db = cachedDb;
     } else {
-      const client = await MongoClient.connect(this.uri);
+      const client = await MongoClient.connect(this.uri, {
+        useUnifiedTopology: true,
+      });
       this.db = client.db(this.dbName);
     }
   }
 
   async findNotifications(frequency) {
     if (frequency === EmailFrequency.INSTANT) {
       return this._findInstantNotifications();
+    } else if (frequency === EmailFrequency.MESSAGE) {
+      return this._findUnreadDirectMessages();
     }
     return this._findDigestNotifications(frequency);
   }
 
-  async setEmailSentAt(notificationIds, frequency) {
+  async _findUnreadDirectMessages() {
+    const threads = await this.db
+      .collection("threads")
+      .aggregate([
+        {
+          $match: {
+            "participants.newMessages": { $gt: 0 },
+            "participants.emailSent": false,
+            "participants.status": {
+              $in: [MessageThreadStatus.ACCEPTED, MessageThreadStatus.PENDING],
+            },
+            $or: [
+              {
+                "participants.lastAccess": null,
+              },
+              {
+                $and: [
+                  {
+                    "participants.lastAccess": {
+                      $lt: DateHelper.subtractMinutes(
+                        new Date(),
+                        this.instantUnreadLookbackInterval,
+                      ),
+                    },
+                  },
+                  {
+                    "participants.lastAccess": {
+                      $gt: DateHelper.subtractMinutes(new Date(), 30),
+                    },
+                  },
```

Comment:
Do we need to keep this? sice we already have 
```js
"participants.newMessages": { $gt: 0 },
"participants.emailSent": false,
```
So it will not send to old threads except if they had newMessages > 0, and it will only send once so it's good, since `emailSent` will be true after that

Reply:
We don't technically need it but I think we should leave it in as a safety, so that we don't send out emails for very old messages. I've developed a notification lambda at work where it started sending out emails for years-old messages due to a bug (because we were not limiting the lookback period in the query), and it was a fiasco. So it would be better to leave it in here. We could increase this lookback if we want though, but I probably wouldn't set this to more than a day.